src/test/java/org/opensearch/ad/TestHelpers.java

```diff
@@ -392,6 +387,61 @@ public static AnomalyDetector randomAnomalyDetectorUsingCategoryFields(String de
         );
     }
 
+    public static AnomalyDetector randomAnomalyDetectorUsingCategoryFields(
+        String detectorId,
+        String timeField,
+        List<String> indices,
+        List<String> categoryFields
+    ) throws IOException {
+        return new AnomalyDetector(
+            detectorId,
+            randomLong(),
+            randomAlphaOfLength(20),
+            randomAlphaOfLength(30),
+            timeField,
+            indices,
+            ImmutableList.of(randomFeature(true)),
+            randomQuery(),
+            randomIntervalTimeConfiguration(),
+            new IntervalTimeConfiguration(0, ChronoUnit.MINUTES),
+            randomIntBetween(1, AnomalyDetectorSettings.MAX_SHINGLE_SIZE),
+            null,
+            randomInt(),
+            Instant.now(),
+            categoryFields,
+            randomUser(),
+            null
+        );
+    }
+
+    public static AnomalyDetector randomAnomalyDetectorUsingCategoryFields(
```

Comment:
nit: multiple `randomAnomalyDetectorUsingCategoryFields` is a little confusing. Maybe can add w/ result index to this one or with custom indices to the one just above since that differentiates the methods. Also nit picky but these could be refactored into only calling the full `new AnomalyDetector(...)` once by gradually passing more variables like we do in `randomAnomalyDetector` if we choose to keep same name.

Reply:
Good catch, will change to cascading constructors.